crates/nu-command/src/bytes/starts_with.rs

```diff
@@ -60,63 +60,13 @@ impl Command for BytesStartsWith {
             pattern,
             cell_paths,
         };
-
-        match input {
-            PipelineData::ExternalStream {
-                stdout: Some(stream),
-                span,
-                ..
-            } => {
-                let mut i = 0;
-
-                for item in stream {
-                    let byte_slice = match &item {
-                        // String and binary data are valid byte patterns
-                        Ok(Value::String { val, .. }) => val.as_bytes(),
```

Comment:
Removing this will be a breaking change, as this worked before:

```nushell
^echo foo | bytes starts-with ("foo" | into binary)
```

In general, I think untyped byte streams should be able to be treated as binary

Reply:
That was me being ~~lazy~~ making the PR smaller :smile:. After this, I intend to open another PR to restore that command and also make `bytes ends-with` support byte streams properly.